ssetests/testapi_sse_client.go

```diff
@@ -0,0 +1,260 @@
+package ssetests
```

Comment:
The logic in this file for telling the test service to create a client, and then receiving callback messages from the client, still basically works the same. I've just moved things around to make it so tests can access this component directly instead of through wrapper methods on `T`.

I also used the "varargs that provide configuration changes" pattern that I used in `sdk-test-harness` to make it easy to set up the interconnected components in the test without having to either pass a lot of parameters explicitly, or rely on hidden state magic in `T`. In `testapi_stream_server` you'll see that `StreamServer` implements the `SSEClientConfigurer` interface. So, after a test has created a `StreamServer`, it can simply pass that component as a config parameter to `NewSSEClient` and automatically get the desired behavior of making the SSE client point to the server's base URL.